core/src/main/java/org/jboss/logmanager/handlers/DelayedHandler.java

```diff
@@ -198,8 +217,9 @@ public final boolean isActivated() {
     private synchronized void activate() {
         // Always attempt to drain the queue
         ExtLogRecord record;
+        final LogContext logContext = this.logContext;
         while ((record = logRecords.pollFirst()) != null) {
-            if (isEnabled() && isLoggable(record)) {
+            if (isEnabled() && isLoggable(record) && (logContext == null || logContext.getLogger(record.getLoggerName()).isLoggable(record.getLevel()))) {
```

Comment:
My concern here would be if this handler was being used during configuration we could lose log messages if the logger was later configured to enable a higher level.

Reply:
In our usage, we set the handler level (and the root logger level) to ALL until we're initialized, so we *want* to lose the log messages.  Better to lose them late because the user doesn't want them, than to lose them early only to discover they were wanted after all.

Reply:
If a `null` `LogContext` is sent in, then the behavior is not changed compared to today.